From 1fcd2c937b1b7a2bc748f9828d242f7031d31c28 Mon Sep 17 00:00:00 2001
From: Anas Nashif <anas.nashif@intel.com>
Date: Fri, 16 May 2025 17:58:04 -0400
Subject: [PATCH] Revert "target/xtensa: Make use of 'segment' in pptlb helper
 less confusing"

This reverts commit b42ba4ea4322357fcbfcb0e6e613d79ede84de64.
---
 target/xtensa/mmu_helper.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/target/xtensa/mmu_helper.c b/target/xtensa/mmu_helper.c
index 29b84d5dbf..997b21d389 100644
--- a/target/xtensa/mmu_helper.c
+++ b/target/xtensa/mmu_helper.c
@@ -991,7 +991,7 @@ uint32_t HELPER(rptlb1)(CPUXtensaState *env, uint32_t s)
 uint32_t HELPER(pptlb)(CPUXtensaState *env, uint32_t v)
 {
     unsigned nhits;
-    unsigned segment;
+    unsigned segment = XTENSA_MPU_PROBE_B;
     unsigned bg_segment;
 
     nhits = xtensa_mpu_lookup(env->mpu_fg, env->config->n_mpu_fg_segments,
@@ -1005,7 +1005,7 @@ uint32_t HELPER(pptlb)(CPUXtensaState *env, uint32_t v)
         xtensa_mpu_lookup(env->config->mpu_bg,
                           env->config->n_mpu_bg_segments,
                           v, &bg_segment);
-        return env->config->mpu_bg[bg_segment].attr | XTENSA_MPU_PROBE_B;
+        return env->config->mpu_bg[bg_segment].attr | segment;
     }
 }
 
-- 
2.43.0

